src/components/ui4/MarketsCardUi4.tsx

```diff
@@ -0,0 +1,137 @@
+import { div, lt } from 'biggystring'
+import * as React from 'react'
+import { View } from 'react-native'
+import FastImage from 'react-native-fast-image'
+
+import { getSymbolFromCurrency } from '../../constants/WalletAndCurrencyConstants'
+import { useAsyncEffect } from '../../hooks/useAsyncEffect'
+import { formatFiatString } from '../../hooks/useFiatText'
+import { useHandler } from '../../hooks/useHandler'
+import { toPercentString } from '../../locales/intl'
+import { getDefaultFiat } from '../../selectors/SettingsSelectors'
+import { asCoinranking, CoinRankingData } from '../../types/coinrankTypes'
+import { useSelector } from '../../types/reactRedux'
+import { NavigationBase } from '../../types/routerTypes'
+import { debugLog, LOG_COINRANK } from '../../util/logger'
+import { fetchRates } from '../../util/network'
+import { DECIMAL_PRECISION } from '../../util/utils'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { EdgeText } from '../themed/EdgeText'
+import { CardUi4 } from './CardUi4'
+import { RowUi4 } from './RowUi4'
+
+const LISTINGS_REFRESH_INTERVAL = 30000
+type Timeout = ReturnType<typeof setTimeout>
+
+interface Props {
+  navigation: NavigationBase
+  numRows: number
+}
+
+/**
+ * Card that displays balance, deposit/send buttons, and a link to view assets
+ */
+export const MarketsCardUi4 = (props: Props) => {
+  const { navigation, numRows } = props
+
+  const theme = useTheme()
+  const styles = getStyles(theme)
+  const defaultIsoFiat = useSelector(state => `iso:${getDefaultFiat(state)}`)
+  const defaultFiat = useSelector(state => getDefaultFiat(state))
+  const fiatSymbol = React.useMemo(() => getSymbolFromCurrency(defaultFiat), [defaultFiat])
+
+  const [coinRankingDatas, setCoinRankingDatas] = React.useState<CoinRankingData[]>([])
+
+  const timeoutHandler = React.useRef<Timeout | undefined>()
+
+  const handleCoinRow = useHandler((coinRow, index) => {
```

Comment:
Please call this `renderEventRow` instead. We use the word "handle" for things passed to `onWhatever` props.

Also remove the `useHandler` wrapper, for the same reason. Since we call the function immediately as part of the render, instead of passing it to a prop, there is no benefit to memoizing it.